src/System.Net.Sockets/src/System/Net/Sockets/TCPClient.Unix.cs

```diff
@@ -378,10 +378,9 @@ private bool ExclusiveAddressUseCore
                 ShadowOptions so = EnsureShadowValuesInitialized();
                 so._exclusiveAddressUse = value ? 1 : 0;
                 so._exclusiveAddressUseInitialized = true;
-                if (_clientSocket != null)
-                {
-                    _clientSocket.ExclusiveAddressUse = value; // Use setter explicitly as it does additional validation beyond that done by SetOption
-                }
+
+                Socket s = _clientSocket ?? CreateSocket();
```

Comment:
If _clientSocket is null, we should be disposing of the temporary socket that gets created.